docs/design/operator.md

```diff
@@ -0,0 +1,414 @@
+                - node
+      tolerations:
+      - key: storage
+        operator: Exists
+      resources:
+        registrar:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        liveness:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        plugin:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+      pluginVolumes:
+      - name: host-run
+        Volumes:
+          hostPath:
+            path: "/run"
+            type: Directory
+        VolumeMounts:
+          name: ''
+          readOnly: true
+          mountPath: "/run"
+          mountPropagation: Bidirectional
+      kubeletDirPath: "/var/lib/kubelet"
+      imagePullPolicy: IfNotPresent
+    provisioner:
+      priorityClassName: system-cluster-critical
+      labels:
+        app: provisioner
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+      provisionerReplicas: 2
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: storage
+                operator: In
+                values:
+                - node
+      tolerations:
+      - key: storage
+        operator: Exists
+      resources:
+        attacher:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        snapshotter:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        resizer:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        provisioner:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        omapGenerator:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        liveness:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        plugin:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+    liveness:
+      metricsPort: 8000
+    leaderElection:
+      leaseDuration: 100
+      renewDeadline: 100
+      retryPeriod: 10
+    deployCSIAddons: true
+    cephfs:
+      useKernelClient: true
+      kernelMountOptions: ms_mode=secure
+status:
+  phase: Succeeded
+  message: operator config successfully created
```

Comment:
Isn't the common practice to use a `reason` filed instead of a `message` field?
```suggestion
  message: operator config successfully created
```

Reply:
i was referring to know API's for it https://github.com/kubernetes/api/blob/664bdd58a30d9e428a31e7e77713a26424025722/core/v1/types.go#L414-L425.

Reply:
I usually see phase/message/reason in resource Conditions, but it seems to me that Conditions have been going out of vogue. I think this is probable because they are incredibly hard to deal with in code and user scripts compared to having typed fields for status.

Many of the recent sig-storage KEPs use a `someResourceReady: bool` (e.g., `bucketReady: bool`, `snapshotReady: bool`) as well, so that users can know with a boolean whether their resource is ready.

Personally, I don't have a preference between `message` and `reason`. Both are alphabetically close to `phase` which should mean users see the outputs near each other as status items are added.

But I would suggest some sort of `xReady` bool that indicates when config is ready.

Reply:
Yes this is true am still not clear should we go with a bool value to indicate that the driver is running successfully after deployment where i need to ensure that the csi driver is able to serve the request for the user or not as it involves multiple check which are not exposed to the consumer yet. but i would like to keep this an option which we need explore in future, @BlaineEXE i hope that should be fine as well.

Reply:
closing as new issue is opened for it.